tidy3d/components/grid/grid.py

```diff
@@ -119,10 +119,23 @@ def spatial_interp(
 
         if fill_value == "extrapolate" and spatial_dataarray.values.size > 0:
             # filter any values larger/smaller than the original data's max/min.
-            max_val = max(spatial_dataarray.values.ravel())
-            min_val = min(spatial_dataarray.values.ravel())
-            interp_dataarray = interp_dataarray.where(interp_dataarray >= min_val, min_val)
-            interp_dataarray = interp_dataarray.where(interp_dataarray <= max_val, max_val)
+            if np.iscomplexobj(spatial_dataarray.values):
+                re = interp_dataarray.values.real
+                limit = spatial_dataarray.values.real.max()
+                re[re > limit] = limit
+                limit = spatial_dataarray.values.real.min()
+                re[re < limit] = limit
+                im = interp_dataarray.values.imag
+                limit = spatial_dataarray.values.imag.max()
+                im[re > limit] = limit
+                limit = spatial_dataarray.values.imag.min()
+                im[re < limit] = limit
+                interp_dataarray.values[:] = re + 1j * im
+            else:
+                max_val = spatial_dataarray.values.max()
+                min_val = spatial_dataarray.values.min()
+                interp_dataarray = interp_dataarray.where(interp_dataarray >= min_val, min_val)
+                interp_dataarray = interp_dataarray.where(interp_dataarray <= max_val, max_val)
```

Comment:
How about a single function like

```
def cap_array(array):
    max_val = spatial_dataarray.values.max()
    min_val = spatial_dataarray.values.min()
    capped_array = array.where(array >= min_val, min_val)
    capped_array = capped_array.where(capped_array <= max_val, max_val)
    return capped_array
```

which is then used three times?

Reply:
There is a function already: `numpy.clip`